api/api.proto

```diff
@@ -1907,6 +1907,43 @@ service OpenStorageFilesystemCheck {
     }
 }
 
+// ## OpenStorageFilesystemDefrag Service
+// This service provides APIs to manage filesystem defragmentation schedules/jobs on a cluster.
+service OpenStorageFilesystemDefrag {
+  // Create a schedule to run defragmentation tasks periodically
+  rpc CreateSchedule(SdkCreateDefragScheduleRequest)
+    returns (SdkCreateDefragScheduleResponse){
+      option(google.api.http) = {
+        post: "/v1/filesystem-defrag/schedule"
+        body: "*"
+      };
+    }
+
+  // Get defrag status of a node
+  rpc GetNodeStatus(SdkGetDefragNodeStatusRequest)
+    returns (SdkGetDefragNodeStatusResponse){
+      option(google.api.http) = {
+        get: "/v1/filesystem-defrag/status/{node_id}"
+      };
+    }
+
+  // Enumerate all nodes, returning defrag status of the entire cluster
+  rpc EnumerateNodeStatus(SdkEnumerateDefragStatusRequest)
+    returns (SdkEnumerateDefragStatusResponse){
+      option(google.api.http) = {
+        get: "/v1/filesystem-defrag/status"
+      };
+    }
+
+  // Clean up all defrag schedules and stop all operations
+  rpc CleanUpSchedules(SdkCleanUpDefragSchedulesRequest)
```

Comment:
does this stop and then delete all currently active schedules?

Reply:
Yes, this will stop all active defrag jobs and then delete all defrag schedules (no matter active or not).